src/core/base.jl

```diff
@@ -6,16 +6,58 @@ export
     run_generic_model, build_generic_model, solve_generic_model
 
 ""
-@compat abstract type AbstractPowerModel end
-
-""
 @compat abstract type AbstractPowerFormulation end
 
 ""
 @compat abstract type AbstractConicPowerFormulation <: AbstractPowerFormulation end
 
-""
-type GenericPowerModel{T<:AbstractPowerFormulation} <: AbstractPowerModel
+"""
+```
+type GenericPowerModel{T<:AbstractPowerFormulation}
+    model::JuMP.Model
+    data::Dict{String,Any} 
+    setting::Dict{String,Any}
+    solution::Dict{String,Any}
+    ref::Dict{Symbol,Any} # reference data
+end
+```
+where
+
+* `data` is the original data, usually from reading in a `.json` or `.m` (patpower) file
+* `setting` usually looks something like `Dict("output" => Dict("line_flows" => true))`, and
+* `ref` is a `Dict` with the following keys:
+
+    * `:off_angmin` and `:off_angmax` (see `calc_theta_delta_bounds(data)`),
+    * `:bus` is the set `{(i, bus) in ref[:bus] : bus["bus_type"] != 4}`,
+    * `:gen` is the set `{(i, gen) in ref[:gen] : gen["gen_status"] == 1 && gen["gen_bus"] in keys(ref[:bus])}`,
+    * `:branch` is the set `{(i, branch) in ref[:branch]` satisfying the conditions
+
```

Comment:
more simply put, this is the set of branches that are active in the network, based on the component status values.